libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Combinators.hs

```diff
@@ -25,11 +25,14 @@ errorMess extra mess = error (unlines ("\nGen-time error" : (reverse (extra : me
 
 -- | suchThat version that tracks Gen-time errors
 suchThatErr :: [String] -> Gen a -> (a -> Bool) -> Gen a
-suchThatErr msgs gen p = do
-  x <- suchThatMaybe (resize 1000 gen) p
-  case x of
-    Just y -> pure y
-    Nothing -> errorMess "SuchThat times out" msgs
+suchThatErr msgs gen p = sized $ \n -> try (10 :: Int) n
+  where
+    try 0 _ = errorMess "SuchThat times out" msgs
+    try k sz = do
+      x <- resize sz $ suchThatMaybe gen p
+      case x of
+        Just y -> pure y
+        Nothing -> try (k - 1) (sz + 5)
 
```

Comment:
No particularly strong reasons. The size varies between 0 and 99 during a `quickCheck` run, so bumping by 5 for each retry seemed reasonable. You want to avoid the size getting too big, because it's used for things like lengths of lists and other datastructures.